models/mli/model_gam.py

```diff
@@ -0,0 +1,242 @@
+"""Generalized Additive Model"""
+
+import uuid
+import os
+import datatable as dt
+import numpy as np
+from h2oaicore.models import CustomModel
+from sklearn.preprocessing import LabelEncoder
+from h2oaicore.systemutils import physical_cores_count
+from h2oaicore.systemutils import temporary_files_path, remove, config
+from h2oaicore.systemutils import make_experiment_logger, loggerinfo, loggerwarning, loggerdebug
+
+
+class GAM(CustomModel):
+    _regression = True
+    _binary = True
+    _multiclass = False
+    _display_name = "GAM"
+    _description = "Generalized Additive Model"
+    _modules_needed_by_name = ['shap', 'collections', 'scipy', 'pandas', 'matplotlib', 'sklearn', 'pygam']
+    _testing_can_skip_failure = False  # ensure tested as if shouldn't fail
+
+    @staticmethod
+    def do_acceptance_test():
+        return True
+
+    def set_default_params(self, accuracy=None, time_tolerance=None,
+                           interpretability=None, **kwargs):
+        # Fill up parameters we care about
+        self.params = dict(random_state=kwargs.get("random_state", 1234),
+                           max_depth_duplication = None, n_estimators= 10,
+                           lam=0.1, max_iter=100)
+        
+    def mutate_params(self, accuracy=10, **kwargs):
+        
+        if accuracy > 8:
+            lam = [0, 0.001, 0.01, 0.1, 1.0, 3.0, 5.0, 10.0]
+            max_iter = [100, 1000]
+
+        elif accuracy >= 5:
+            lam = [0,  0.01, 0.1, 1.0, 10.0]
+            max_iter = [100]
+
+        else:
+            lam = [0,  0.01, 0.1, 1.0, 10.0]
+            max_iter = [100]
+
```

Comment:
Is there reasoning behind parameters selected for these models? Maybe good to comment.